k8srm-prototype/pkg/api/claim_types.go

```diff
@@ -176,150 +205,64 @@ type DeviceClaimStatus struct {
 	// ClassConfigs contains the entire set of dereferenced vendor
 	// configurations from the DeviceClass, as of the time of allocation.
 	// +optional
+	// +listType=atomic
 	ClassConfigs []runtime.RawExtension `json:"classConfigs,omitempty"`
 
 	// ClaimConfigs contains the entire set of dereferenced vendor
 	// configurations from the DeviceClaim, as of the time of allocation.
 	// +optional
+	// +listType=atomic
 	ClaimConfigs []runtime.RawExtension `json:"claimConfigs,omitempty"`
 
 	// Allocations contains the list of device allocations needed to
 	// satisfy the claim, one per pool from which devices were allocated.
 	//
 	// Note that the "current capacity" of the cluster is the result of
 	// applying all such allocations to the published DevicePools. This
-	// means storing these allocations only in claim status fields is likely
-	// to scale poorly, and we will need a different strategy in the real
-	// code. For example, we may need to accumulate these in the DevicePool
-	// status fields themselves, and just reference them from here.
+	// means storing these allocations only in claim status fields is
```

Comment:
Yeah, I can remove this comment. Premature optimization. My concern is just at scheduler startup, it probably needs to process all of these into an accumulation cache so it can make faster decisions. But that's probably pretty fast. The numbers here are frankly quite low (100s of thousands, typically).

Reply:
Fixed in latest commit.